ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,731 @@
+            )
+            concentration = concentration_parameters(
+                geom=geom_selected, image=image_selected, hillas_parameters=hillas,
+            )
+            morphology = morphology_parameters(geom=geometry, image_mask=signal_pixels)
+            intensity_statistics = descriptive_statistics(
+                image_selected, container_class=IntensityStatisticsContainer
+            )
+
+            if peak_time is not None:
+                timing = timing_parameters(
+                    geom=geom_selected,
+                    image=image_selected,
+                    peak_time=peak_time[signal_pixels],
+                    hillas_parameters=hillas,
+                )
+                peak_time_statistics = descriptive_statistics(
+                    peak_time[signal_pixels],
+                    container_class=PeakTimeStatisticsContainer,
+                )
+            else:
+                timing = TimingParametersContainer()
+                peak_time_statistics = PeakTimeStatisticsContainer()
+
+            return ImageParametersContainer(
+                hillas=hillas,
+                timing=timing,
+                leakage=leakage,
+                morphology=morphology,
+                concentration=concentration,
+                intensity_statistics=intensity_statistics,
+                peak_time_statistics=peak_time_statistics,
+            )
+
+        # return the default container (containing nan values) for no
+        # parameterization
+        return ImageParametersContainer()
+
+    def _process_events(self, writer):
+        self.log.debug("Writing DL1/Event data")
+        is_initialized = False
+        self.event_source.subarray.info(printer=self.log.debug)
+
+        for event in tqdm(
+            self.event_source,
+            desc=self.event_source.__class__.__name__,
+            total=self.event_source.max_events,
+            unit="ev",
+            disable=not self.progress_bar,
+        ):
+
+            if not is_initialized:
+                self._write_simulation_configuration(writer, event)
+                is_initialized = True
+
+            self.log.log(9, "Writing event_id=%s", event.index.event_id)
+
+            self.calibrate(event)
+
+            event.mc.prefix = "mc"
+            event.trig.prefix = ""
+            self._cur_obs_id = event.index.obs_id
+
+            # write the subarray tables
+            writer.write(
+                table_name="simulation/event/subarray/shower",
+                containers=[event.index, event.mc],
+            )
+            writer.write(
+                table_name="dl1/event/subarray/trigger",
+                containers=[event.index, event.trig],
+            )
+            # write the telescope tables
+            self._write_telescope_event(writer, event)
+
+        if is_initialized is False:
+            raise ValueError(f"No events found in file: {self.event_source.input_url}")
```

Comment:
Do we really want to raise on empty runs? I don't know how this is for cta mc productins, but at least in FACT we had some Proton simulation runs where no event triggered. Why would you raise on successfully processing an empty file? THat's valid information, no?

Reply:
Perhaps a warning would be better. It's there becasue if you for example specify a list of telescopes [1,2,3,4], but there are no events that trigger those telescopes, there should be a good error message (I added it since I ran into that problem and was confused why I had no output).   Also rewording it to "no events found in the file for the telescope ids [xxxx]" would be clearer.

Reply:
basically it's to explain to the user that they misconfigured the telescope list or something and excluded all events.